addon/components/nypr-account-forms/set-password.js

```diff
@@ -0,0 +1,63 @@
+import layout from '../../templates/components/nypr-account-forms/set-password';
+import Component from 'ember-component';
+import get from 'ember-metal/get';
+import set from 'ember-metal/set';
+import computed from 'ember-computed';
+import service from 'ember-service/inject';
+import Changeset from 'ember-changeset';
+import lookupValidator from 'ember-changeset-validations';
+import PasswordValidations from 'nypr-account-settings/validations/nypr-accounts/password';
+import RSVP from 'rsvp';
+import fetch from 'fetch';
+import { rejectUnsuccessfulResponses } from 'nypr-account-settings/utils/fetch-utils';
+
+export default Component.extend({
+  layout,
+  store: service(),
+  authAPI: null,
+  session: null,
+  resendUrl: computed('authAPI', function() {
+    return `${get(this, 'authAPI')}/v1/password/forgot`;
+  }),
+  allowedKeys: ['password'],
+  passwordExpired: false,
+  passwordWasSet: false,
+  init() {
+    this._super(...arguments);
+    set(this, 'fields', {
+      password: ''
+    });
+    set(this, 'changeset', new Changeset(get(this, 'fields'), lookupValidator(PasswordValidations), PasswordValidations));
+    get(this, 'changeset').validate();
+  },
+  setPassword(username, email, temp, new_password) {
```

Comment:
Ah I based this code off of the reset password stuff where I could just do

```resetPassword(email, new_password, confirmation) {
    let body = JSON.stringify({email, new_password, confirmation});```

Doesn't make any sense here though. Probably really shouldn't do it there either. I'll update this.